base/replicator.go

```diff
@@ -52,7 +52,16 @@ func (r *Replicator) Replicate(params sgreplicate.ReplicationParameters, isCance
 		return nil, r.stopReplication(replicationId)
 
 	} else {
-		// Check whether specified replication is already active
+		replicationId := params.ReplicationId
+		// If replicationId is defined, check that a replication with the same ID is not already running
+		if replicationId != "" {
+			sgreplication := r.getReplication(replicationId)
+			if  sgreplication != nil {
+				return nil, HTTPErrorf(http.StatusConflict, "Replication already active for specified replication_id [%v]",replicationId)
+			}
+		}
+
+		// Check whether specified replication is already active for the given params
```

Comment:
There seems to be some redundant code here to identify a replication id based on params.  What do you think about doing the following:
1. Move the replicationId := params.ReplicationId and if replicationId = "" handling down into getReplicationForParams
2. Call replication, found := r.getReplicationForParams(params) once, before we do the isCancel check.

I think we lose some granularity on the error messages ("...specified replication_id" vs "specified parameters"), but I think that's fine.